### DISCUSSION TITLE
Cannot stream results of `INSERT INTO ... RETURNING` from PostgreSQL

### DISCUSSION THREAD

**Discussion by Prometheus3375** [sqlalchemy/sqlalchemy#12127] [Usage Questions]
Here is a simple code which adds cargo type `Other` to the table `cago_types`.

```python
engine = create_async_engine(
        # URI: postgresql+psycopg://postgres:REDACTED@localhost:5432/postgres
        settings.database_uri,
        pool_size=settings.pool_size,
        max_overflow=settings.pool_overflow,
        pool_recycle=settings.pool_recycle,
        pool_timeout=settings.pool_timeout,
        )

session_maker = async_sessionmaker(engine, autobegin=False)


class CargoTypeTable(DeclarativeBase):
    __tablename__ = 'cargo_types'

    id: Mapped[int] = mapped_column(BigInteger, autoincrement=True, primary_key=True)
    name: Mapped[str] = mapped_column(String(50), unique=True, nullable=False)


async with session_maker.begin() as session:
    vals = [dict(name='Other')]
    query = (
            insert(CargoTypeTable)
            .values(vals)
            .on_conflict_do_nothing()
            .returning(CargoTypeTable)
        )
    result = await session.scalars(query)
    for row in result:
        print(f'Added cargo type {row.name!r}')
```

This works fine, but I want to stream returning results:
```python
    result = await session.stream_scalars(query)
    async for row in result:
        print(f'Added cargo type {row.name!r}')
```

However, calling `stream_scalars` causes an error:
```
sqlalchemy.exc.ProgrammingError: (psycopg.errors.SyntaxError) syntax error at or near "INSERT"
LINE 1: DECLARE "c_16bf10022a0_1" CURSOR FOR INSERT INTO cargo_types...
                                             ^
[SQL: INSERT INTO cargo_types (name) VALUES (%(name_m0)s::VARCHAR) ON CONFLICT DO NOTHING RETURNING cargo_types.id, cargo_types.name]
[parameters: {'name_m0': 'Other'}]
```

Why I am getting this error? I found similar issue on [StackOverflow](https://stackoverflow.com/questions/58347638/sqlalchemy-programmingerror-when-inserting-into-postgresql-11-table), but the solution was to disable streaming which exactly what I want to do.

**Comment by CaselIT** [comment]
Hi,

Declare is generated by psycopg, so it's probably best if you ask for suggestions there. The statement generated by sqlalchemy is `SQL: INSERT INTO cargo_types (name) VALUES (%(name_m0)s::VARCHAR) ON CONFLICT DO NOTHING RETURNING cargo_types.id, cargo_types.name
`

**Reply by Prometheus3375** [reply]
Hm, interesting, I will investigate this. I was using `psycopg` directly with streaming afaik and never had such issue.

**Reply by CaselIT** [reply]
Not sure what is happening, feel free to link that discussion here if you open one.

It seems that this bit `DECLARE "c_16bf10022a0_1" CURSOR FOR ` is added by psycopg
